src/main/java/teammates/storage/api/FeedbackResponsesDb.java

```diff
@@ -572,7 +591,7 @@ public void updateFeedbackResponse(FeedbackResponseAttributes newAttributes)
      * Preconditions: <br>
      * * {@code newAttributes.getId()} is non-null and correspond to an existing feedback response.
      */
-    public void updateFeedbackResponseOptimized(FeedbackResponseAttributes newAttributes, FeedbackResponse fr) 
+    public void updateFeedbackResponseOptimized(FeedbackResponseAttributes newAttributes, FeedbackResponse fr, boolean keepUpdateTimestamp) 
```

Comment:
See if some of these `updateFeedbackResponse` methods can be private. I have a feeling you won't need them all to be public, at least now.